OOP_Lesson5/Task2.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * Сделайте класс Student со свойствами $name и $course (курс студента, от 1-го до 5-го). 
+ * В классе Student сделайте public метод transferToNextCourse, который будет переводить студента на следующий курс.
+ * Выполните в методе transferToNextCourse проверку на то, что следующий курс не больше 5.
+ * Вынесите проверку курса в отдельный private метод isCourseCorrect 
+ */
+/**
+ * [Description Student]
+ */
+class Student
+{
+    private string $name;
+    private int $course;
+    /**
+     * @param string $name
+     * @param int $course
+     */
+    public function __construct(string $name, int $course) 
+    {
+        
+        $this->name = $name;
+        if ($this->isCourseCorrect($course)) {
+            $this->course = $course;
+        } else {
+            $this->course = 1;
+        }
+       
+    }
+    /**
+     * @return string
+     */
+    public function getName() : string 
+    {
+        return $this->name;
+    }
+    /**
+     * @return int
+     */
+    public function getCourse() : int 
+    {
+        return $this->course;
+    }
+    /**
+     * @return Student
+     */
+    public function transferToNextCourse() : Student
+    {
+        $newCourse = $this->course + 1;
+        if ($this->isCourseCorrect($newCourse)) {
```

Comment:
In case then false, student remains on the 5th course?

Reply:
Yes, because 5th course is the maximum. There is no option of graduating student, so he stays on the max course which is 5th course